### DISCUSSION TITLE
How to add Pitch info (xDistance, yDistance) mentioned in some lefs in tech json ?

### DISCUSSION THREAD

**Discussion by SupraCoder** [ucb-bar/hammer#727] [Support (Q&A)]
The lef specification supports describing pitch in terms of distance or x/y distances.
Like this: PITCH {distance | xDistance yDistance}.

In tech json, we just have one field to fill this info (distance). How do we add Pitch info if our lef has pitch in terms of 
xDistance and  yDistance ? Should we simply take the bigger value of the two ? Or is there a way to specify both the values ?

**Comment by edwardcwang** [comment]
Are you talking about the pitch as specified in the tech LEF, or for a macro?

**Comment by SupraCoder** [comment]
As specified in the tech Lef.

**Comment by edwardcwang** [comment]
Hammer should pass along the tech LEF to downstream tools automatically, so probably no action is required. For power strap generation though, you might need to specify a pitch for that IIRC.

**Comment by SupraCoder** [comment]
Thank you very much for the response. I think I got what I was looking for.

**Comment by edwardcwang** [comment]
Awesome!

**Comment by colinschmidt** [comment]
Just to follow up, for the power strap settings it doesn't appear that hammer `Metal` layer library supports different x and y pitches so if you wanted to use that  (a very nice feature IMO) you could just conservatively put the larger of the two pitches.

**Comment by SupraCoder** [comment]
Thank you for the info. That is what we decided to do. We are using the larger of the two pitches now. It would be nice to see this feature supported in the tool though.

**Comment by colinschmidt** [comment]
It seems relatively straightforward to add given we already track the orientation of `Metal`s so just a little bit of python if you are interested in becoming a contributor 😄

**Comment by SupraCoder** [comment]
Will give it a try as soon as I am done with what I am working on right now :)
